### PR TITLE
Added iteration attributes to Module, Global, and Function

### PR DESCRIPTION
**Pull request by harlanhaskins**

### CODE REVIEW

File: Sources/LLVM/Module.swift [hunk 1]
```diff
@@ -123,11 +123,64 @@ public final class Module {
...
+  }
+
   /// Dump a representation of this module to stderr.
   public func dump() {
     LLVMDumpModule(llvm)
   }
 
+  /// The full text IR of this module
+  public var description: String {
+    let cStr = LLVMPrintModuleToString(llvm)!
```
Comment (CodaFi) [inline Sources/LLVM/Module.swift:179]: This seems... heavy for `description`.  But I can find precedent in other bindings for this.

**Comment by CodaFi** [comment]
LGTM modulo the discussion I just started.

File: Sources/LLVM/Module.swift (hunk 1 above)
Comment (harlanhaskins) [reply Sources/LLVM/Module.swift:179]: I can't think of any other thing for `description`.
This would also allow for `print(module)` to print the IR.

Comment (CodaFi) [reply Sources/LLVM/Module.swift:179]: :shipit:

**Review by CodaFi — APPROVED** [review]

**Pull request merged**